java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowStreamReader.java

```diff
@@ -170,4 +172,25 @@ protected ArrowDictionaryBatch readDictionary() throws IOException {
 
     return MessageSerializer.deserializeDictionaryBatch(result.getMessage(), bodyBuffer);
   }
+
+  @Override
+  protected void readDictionaries(Map<Long, Dictionary> dictionaries) throws IOException {
+    // Read and load all dictionaries from schema
+    for (int i = 0; i < dictionaries.size(); i++) {
+      ArrowDictionaryBatch dictionaryBatch = readDictionary();
+      if (dictionaryBatch == null) {
+        // empty stream has no dictionaries in IPC.
+        if (i == 0) {
+          initialized = true;
+          if (loadNextBatch()) {
+            throw new IOException("DictionaryBatch is missing.");
+          }
+          break;
+        } else {
+          throw new IOException("Unexpected end of input. Expected DictionaryBatch");
```

Comment:
It would be potentially useful to include "i" in the error message.

Reply:
Sure, fixed.